### PR TITLE
Upgrade to egui v0.29.1 and glium v0.36.0

### PR DESCRIPTION
**Pull request by NekomaruQwQ**
This pull request upgrades the following dependencies to the latest:
 - `egui` to 0.29.1
 - `glium` to 0.36.0
 - `image` to 0.25.4 (dev dependency for the `native_texture` example) 

This pull request also includes a rearrange on the imports for all code to make the style more consistent and self-explanative. If such change is not appreciated, I'd open another pull request containing the dependency upgrades only.

Btw, if this crate is looking for some kind of active maintainer, I'm more than willing to help. I'm a graphics dev with experience of authoring and active maintaining `egui-directx11`.

### CODE REVIEW

**Comment by fayalalebrun** [comment]
Yes, please move the style changes to another PR so we can discuss them separately.

Regarding becoming a maintainer, I am very much open to the idea after you have the chance to contribute a bit more.

**Comment by NekomaruQwQ** [comment]
I've reverted all my style changes preserving only changes required for the dependency upgrades. Note that my commit is based on @skillissuedev's work, which contains some style changes that I'd not revert and these changes remain in the overall diffs.

P.S. I force pushed to avoid opening another pull request and I only force push in my own temporary repos.

File: Cargo.toml
Comment (fayalalebrun) [inline Cargo.toml:34]: It seems like the `pure_glium` example is broken at least on Wayland. The fonts are missing. You need to enable the `default_fonts` feature of `egui` for the example to work correctly. Though preferably this feature would only be enabled by default in dev.

**Review by fayalalebrun — CHANGES_REQUESTED** [review]

File: examples/native_texture.rs
Comment (bschwind) [inline examples/native_texture.rs:29]: I was dumb and did all the same work of updating the dependencies before I saw this PR. Though if we want to not add this `#[allow]`, we could pull parts of #4 in as a followup PR so it follows the new winit's "best practices".

**Comment by NekomaruQwQ** [comment]
I've appended a new commit that enables the `default_fonts` feature of `egui` as a dev dependency regarding the missing font in examples issue mentioned by @fayalalebrun, tested on Windows. Please review and check if there's any other issue in this PR.

Btw, what do you think of the idea for this crate to depend on the re-exported `egui` and `winit` from `egui-winit` rather than depending directly on them? @skillissuedev introduced this approach and I kept it as I found that it reasonably simplifies for dependency management. I'm a little bit concerned about the potential confusion in the future though.

File: examples/native_texture.rs
Comment (NekomaruQwQ) [reply examples/native_texture.rs:29]: That is a huge refactor and I believe that the maintainers would appreciate it. We may let the maintainers decide which PR to work on.

(and what a coincidence that we worked on the same issue at almost the same time XD)

**Comment by NekomaruQwQ** [comment]
And regarding @bschwind's work, our maintainers may want decide which PR to continue with. #4 goes a step further adopting the new trait-based API of `winit`, which we have to follow up one day in the future.

**Comment by bschwind** [comment]
> Btw, what do you think of the idea for this crate to depend on the re-exported egui and winit from egui-winit rather than depending directly on them?

I have a slight preference against doing this in most cases. I used to do this in the past with some projects but in some cases the compiler had a harder time finding auto-import suggestions, and if you want to enable extra feature flags in the transitive dependency it becomes more annoying. Although it's a tiny bit of extra work, I feel like clarity is higher when you're importing the dependencies you need directly from crates.io instead of transitively through another crate.

**Comment by fayalalebrun** [comment]
For now I prefer to continue with #4 since it also addresses the use of deprecated methods in the examples. However, after that has been merged I welcome you to open another MR if you would like to incorporate any additional change from this MR.

**Pull request closed without merging**
